File: Python/Math-Array-String/17.py
# 2833. Furthest Point From Origin


# class Solution:
#     def furthestDistanceFromOrigin(self, moves: str) -> int:
#         L, R, _ = 0, 0, 0
#         max_distance = 0
        
#         for move in moves:
#             if move == 'L':
#                 L += 1
#             elif move == 'R':
#                 R += 1
#             elif move == '_':
#                 _ += 1
            
        
#         if L < R:
#             max_distance = (R + _) - L
#         else:
#             max_distance = (L + _) - R
            
#         return max_distance


class Solution:
    def furthestDistanceFromOrigin(self, moves: str) -> int:
        L = moves.count("L") 
        R = moves.count("R") 
        _ = moves.count("_") 
            
        if L < R:
            return (R + _) - L
        else:
            return (L + _) - R
            

sol = Solution()
output = sol.furthestDistanceFromOrigin("LLRRRR")
print(output)

File: JavaScript/Math-Array-String/17.js
// 2833. Furthest Point From Origin

/**
 * @param {string} moves
 * @return {number}
 */
// var furthestDistanceFromOrigin = function(moves) {
//     let L = moves.split("L").length - 1;
//     let R = moves.split("R").length - 1;
//     let _ = moves.split("_").length - 1;
//     console.log(_, L, R);
    
    
    // if( L < R){
    //     return R + _ - L;
    // }else{
    //     return L + _ - R;
    // }
// };

var furthestDistanceFromOrigin = function(moves) {
    let L = 0, R = 0, _ = 0;

    for (let char of moves) {
        if (char === 'L') L++;
        else if (char === 'R') R++;
        else if (char === '_') _++;
    }

    if( L < R){
        return R + _ - L;
    }else{
        return L + _ - R;
    }
};


console.log(furthestDistanceFromOrigin("_RR__LL_"));